web3/contract.py

```diff
@@ -1328,6 +1332,39 @@ def factory(cls, class_name, **kwargs):
         return PropertyCheckingFactory(class_name, (cls,), kwargs)
 
 
+class CallerMethod:
+    def __init__(self, function, normalizers=None):
+        self._function = function
+
+    def __call__(self, *args, **kwargs):
+        # TODO: Remove this and consolidate into ContractCaller.
+        # I don't know why this is working but it is.
+        return self._function(*args, **kwargs).call()
+
+
+class ContractCaller:
+    def __init__(self, abi, web3, address, *args, **kwargs):
```

Comment:
Instead of `*args, **kwargs` I think it would be sufficient to have a single `transaction_dict=None`.

Reply:
I was thinking that you could optionally pass in a plain old dictionary without the `transaction` keyword, in which case I'd need to keep *args. Here is an example: https://web3py.readthedocs.io/en/stable/contracts.html#web3.contract.ContractFunction.call. Am I reading those docs right or does a transaction dict always need the `transaction` keyword in front of it?

Reply:
I can't dive in right this second to make sure it works, but what I was thinking in general is that you can use the fact that positional args can fill in keyword args, like:

```py
def fn(kw1=None, **kwargs):
  return kw1

assert fn(1) == 1
```